guides/index_lifecycle.md

```diff
@@ -0,0 +1,157 @@
+```
+
+## Index API Actions
+
+### Create a new index
+
+You can quickly create an index with default settings and mappings by using the `indices.create` API action. The following example creates an index named `paintings` with default settings and mappings:
+
+```python
+client.indices.create(index='paintings')
+```
+
+To specify settings and mappings, you can pass them as the `body` of the request. The following example creates an index named `movies` with custom settings and mappings:
+
+```python
+client.indices.create(
+    index='movies',
+    body={
+        'settings': {
+            'index': {
+                'number_of_shards': 2,
+                'number_of_replicas': 1
+            }
+        },
+        'mappings': {
+            'properties': {
+                'title': {'type': 'text'},
+                'year': {'type': 'integer'}
+            }
+        }
+    }
+)
+```
+
+When you create a new document for an index, OpenSearch will automatically create the index if it doesn't exist:
+
+```python
+print(client.indices.exists(index='burner'))  # => False
+client.create(index='burner', body={'lorem': 'ipsum'})
+print(client.indices.exists(index='burner'))  # => True
+```
+
+### Update an Index
+
+You can update an index's settings and mappings by using the `indices.put_settings` and `indices.put_mapping` API actions.
+
+The following example updates the `movies` index's number of replicas to `0`:
+
+```python
+client.indices.put_settings(
+      index='movies',
+      body={
+            'index': {
+            'number_of_replicas': 0
+            }
+      }
+)
+```
+
+The following example updates the `movies` index's mappings to add a new field named `director`:
+
+```python
+client.indices.put_mapping(
+    index='movies',
+    body={
+        'properties': {
+            'director': {'type': 'text'}
+        }
+    }
+)
+```
+
+### Get Metadata for an Index
+
+Let's check if the index's settings and mappings have been updated by using the `indices.get` API action:
+
+```python
+print(client.indices.get(index='movies'))
+```
+
+The response body contains the index's settings and mappings:
+
+```python
+{
+  "movies": {
+    "aliases": {},
+    "mappings": {
+      "properties": {
+        "title": {"type": "text"},
+        "year": {"type": "integer"},
+        "director": {"type": "text"}
+      }
+    },
+    "settings": {
+      "index": {
+        "creation_date": "1680297372024",
+        "number_of_shards": "2",
+        "number_of_replicas": "0",
+        "uuid": "FEDWXgmhSLyrCqWa8F_aiA",
+        "version": {"created": "136277827"},
+        "provided_name": "movies"
+      }
+    }
+  }
+}
+```
+
+### Delete an Index
+
+Let's delete the `movies` index by using the `indices.delete` API action:
+
+```python
+client.indices.delete(index='movies')
+```
+
+We can also delete multiple indices at once:
+
+```python
+client.indices.delete(index=['movies', 'paintings', 'burner'], ignore=[404])
```

Comment:
`ignore` is not an option for python. So this would be:

```
print(client.indices.delete(index=['paintings', 'burner']))
```

since `movies` is deleted in the previous line.

Reply:
But other guides for clients have the presence of ignoring. This is part of the guide. 
Did I make a mistake or is it just that there is no option to ignore the 404 error?

Reply:
But other guides for clients have the presence of ignoring. This is part of the guide.
Did I make a mistake or is it just that there is no option to ignore the 404 error?

Reply:
Looking at the [code](https://github.com/opensearch-project/opensearch-py/blob/03b72c3c929445b979560e5a2d730a83c0ba35d5/opensearchpy/client/indices.py#L295), the parameter available for ignoring non-existing indices is `ignore_unavailable`. I tried it and the following works:

```
print(client.indices.delete(index=['art', 'paintings', 'burner'], ignore_unavailable=True))
```

where `art` index does not exist. The response is:

```
{'acknowledged': True}
```

Lets change it to `ignore_unavailable`.

Reply:
Yes, thanks for the correction! I had my doubts, as the rust client also had this option.